core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -256,19 +251,41 @@ class ZkUtils(val zkClient: ZkClient,
   }
 
   /**
-   * Register brokers with v2 json format (which includes multiple endpoints).
+   * Register brokers with v3 json format (which includes multiple endpoints and rack).
```

Comment:
Right, so that was my original question, this helps with people who want to upgrade brokers from 0.9.0.0 to 0.10.0.0, right? That's a fair argument and worth mentioning in the code. With this approach, only old clients have to go to 0.9.0.1 before the 0.10.0.0 upgrade. Brokers can go straight from 0.9.0.0 to 0.10.0.0.